pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -932,8 +935,52 @@ private void increaseAvailablePermits(ClientCnx currentCnx, int delta) {
         }
     }
 
+    private ByteBuf decryptPayloadIfNeeded(MessageIdData messageId, MessageMetadata msgMetadata, ByteBuf payload,
+            ClientCnx currentCnx) {
+
+        if (msgMetadata.getEncryptionKeysCount() == 0) {
+            return payload.retain();
```

Comment:
Is there any reason to `retain()` payload in this method? because I don't see it released for this retain()?

Reply:
Seems unnecessary as well.

Reply:
I had to put it back since encryptedPayload is released explicitly.